theano/gof/op.py

```diff
@@ -784,6 +784,21 @@ class Op(utils.object2, PureOp, CLinkerOp):
     Convenience class to bundle `PureOp` and `CLinkerOp`.
 
     """
+
+    # We add a default get_params() implementation which will try to detect params from the op
+    # if params_type is set to a Wrapper. If not, we raise a MethodNotDefined exception.
+    def get_params(self, node):
+        if hasattr(self, 'params_type') and isinstance(self.params_type, theano.gof.wrapper.Wrapper):
+            wrapper = self.params_type
+            if hasattr(self, '__props__') and all(field in self.__props__ for field in wrapper.fields):
```

Comment:
Based on @nouiz 's comments above, here: https://github.com/Theano/Theano/pull/5612#discussion_r103782532 , and here: https://github.com/Theano/Theano/pull/5612#issuecomment-282417090 .